### ISSUE TITLE
Error: Could not find function "pandoc_latex_engine_args"

### ISSUE THREAD

**Issue by M-E-Steiner** [hadley/ggplot2-book#104]
When i press STRG+SHIFT+B in RStudio to build the book, i get this error 

    mkdir -p book/tex
    mkdir -p book/tex/_figures
    mkdir -p book/tex/diagrams
    cp book/ggplot2-book.tex book/tex/ggplot2-book.tex
    Rscript book/render-tex.R data-manip.rmd
    Error in rmarkdown::pandoc_options(to = "latex", from = "markdown_style",  : 
       could not find function "pandoc_latex_engine_args"
    Calls: <Anonymous> -> <Anonymous> -> structure -> <Anonymous>...

I have `R 3.2.2`, `Rtools 3,2,x` , `pandoc (1.15.2)` and installed all the dependencies as well as the `inconsolata` font. I have `MikTeX 2.9` and also updated all packages that where updatable. 

When I try `?pandoc_latex_engine_args` the help page opens so this function should be available to R. I tried on 2 different computers (Windows 10 and Windows 7)  and reran the complete installation process on each without any change. I also tried to manually build the book using the `Cygwin` terminal but this gives the same error.

**Comment by neuwirthe** [comment]
Adding 

    library(rmarkdown) 

to book/render-tex.R solved this problem, but then I get

    pandoc: Unknown reader: markdown_style Error: pandoc document conversion failed with error 7 
    Execution halted make: *** [book/tex/data-manip.tex] Error 1

    Exited with status 2.

**Comment by M-E-Steiner** [comment]
Its the same for me now. After inserting `library(rmarkdown)` The new error is now:

    pandoc: Unknown reader: markdown_style Error
    Error: pandoc document conversion failed with error 7

**Comment by neuwirthe** [comment]
I changed book/render-tex.T even more:

    library("methods") # avoids weird broom error
    library("rmarkdown")
    # the line above has been added
    
    path <- commandArgs(trailingOnly = TRUE)
    # command line args should contain just one chapter name
    if (length(path) == 0) {
      message("No input supplied")  
    } else {
      base <- bookdown::tex_chapter()
      base$knitr$opts_knit$width <- 69
      base$pandoc$from <- "markdown"
    # the line above has been added
    
     rmarkdown::render(path, base, output_dir = "book/tex", envir = globalenv(), quiet = TRUE)
    }


This produces the TeX files.
I hat to install a few additional missing packages, and directlabels packages from CRAN did not work, but the one from github (tdhock/directlabels) worked.

Now I can produce all the tex files, but TeXing them breaks.
Two main reasons:
Inconsolata is not found. I have read some posts about this problem and fiddled with ggplot-book.tex
by changing

    \setmonofont[Mapping=tex-ansi]{Inconsolata}

and using

    \usepackage{inconsolata}

but I have not been able to make this work.
Additionally, TeX complains about missing references.

I am running R 3.2.2 on OSX.

**Comment by M-E-Steiner** [comment]
Ok that gets us even closer. When I include your proposed changes to **book/render-tex.R** it seems like all .tex files get build but after the last build i get

    Rscript book/render-tex.R toolbox.rmd
    cp -R _figures/* book/tex/_figures
    cp -R diagrams/* book/tex/diagrams
    # strip bad ICC metadata
    find book/tex -type f -name "*.png" -exec optipng -strip all -o0 -clobber -quiet {} \;
    find: 'optipng': No such file or directory
    find: 'optipng': No such file or directory
    .
    .
    .

this `find: 'optipng': No such file or directory` I get like 100 times until it finishes with

    cp -R book/springer/* book/tex
    cp book/latexmk book/tex/
    cp book/latexmkrc book/tex/
    cd book/tex && ./latexmk -xelatex -interaction=batchmode ggplot2-book.tex
    ./latexmk: permission denied
     make: *** [book/ggplot2-book.pdf] Error 126

    Exited with status 2.

When i try to compile **book/tex/ggplot2-book.tex** i also get tons of errors (some of them mentioning `inconsolata`) but it does nevertheless produce a .pdf. This .pdf has the preface and the booktitlepage in it but the rest is missing.

**Comment by neuwirthe** [comment]
I installed optipng with homebrew,

**Comment by neuwirthe** [comment]
I managed!

in ggolot2-book.tex made some changes:

    \usepackage{inconsolata}
    %% the above line as been added
    \defaultfontfeatures{Mapping=tex-text,Scale=MatchLowercase}
    %% \setmonofont[Mapping=tex-ansi]{Inconsolata}
    %% the above line has been commente out

**Comment by M-E-Steiner** [comment]
nice thank you!, that  works - at least manually. I guess the question now is: do these changes need to be integreated or is this just something that everyone has to go through?

**Comment by suimong** [comment]
@M-E-Steiner May I ask how you solved the "Permission Denied" issue please? I am building ggplot2-book on Windows 8.1 and bumped into almost the exact problems that you listed here, including the final "Permission Denied" one. Thanks to you and @neuwirthe I have come close to complete the build process but I am now stuck at the final step. I have made sure that the book/text/latexmk file, as well as all other files in the project folder, has full read/write/execute permission for everyone on my OS, but the error still persists. This is when I run the build process in RStudio. I then tried to manually run "latexmk -xelatex -interaction=batchmode ggplot2-book.tex" in command prompt and it just worked.

**Comment by M-E-Steiner** [comment]
I actually never really solved the "Permission Denied" issue. As far as I remember I also just manually compiled the `ggplot2-book.tex` file and although I got numerous errors that I didnt really understand I got the pdf in a form that was readable. This was rather dirty but I really wanted to read it so I was happy to have a readable version:-)

**Comment by suimong** [comment]
I see, so basically we did the same thing just to get hold on a readable version as soon as possible.






On Wed, Dec 23, 2015 at 2:45 AM -0800, "Manuel Steiner" <notifications@github.com> wrote:












I actually never really solved the "Permission Denied" issue. As far as I remember I also just manually compiled the ggplot2-book.tex file and although I got numerous errors that I didnt really understand I got the pdf in a form that was readable. This was rather dirty but I really wanted to read it so I was happy to have a readable version:-)

**Comment by hadley** [comment]
Did you run `devtools::install_deps()`? That should get you the right versions of all the packages.

**Comment by suimong** [comment]
Yes I ran `devtools::install_deps()` before building. There are two things I discovered during troubleshooting: 
One is that the mkdir command will introduce a chaos to the ACL of the created folders on Windows, thus causing the "permission denied" error. This can be quickly fixed by hand. However the building process will still throw an "permission denied" error on line 12 of Makefile.
Here is the second discovery: if I change "./latexmk" on line 12 to just "latexmk" then the error disappears and the building procedure will continue.

P.S. I'm building this on Windows 8.1 and Windows Server 2012 R2, same error and same fix.
